### PR TITLE
Sergio ortiz development

### PR DESCRIPTION
**Pull request by Ortiz-Sergio**

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -32,17 +32,28 @@
...
   }
 
-  private void initializeComments() {
-    commentList.add("Awesome Website!");
-    commentList.add("Great job! Keep it up!");
-    commentList.add("Very interesting!!");
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+      String comment = getParameter(request, "comment-input", "");
+      commentList.add(comment);
```
Comment (abhijitnukalapati) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: do we want to filter out empty comments?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -32,17 +32,28 @@
...
+      response.sendRedirect("/index.html");
+  }
+
+  /**
+   * @return the request parameter, or the default value if the parameter
+   *         was not specified by the client
+   */
```
Comment (abhijitnukalapati) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:77]: _minor_ - it seems a bit odd that we have a return block in the comment section with no description. How about just having a description and do away with the `@return`. What do you think?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
Comment (Ortiz-Sergio) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Good idea! Maybe also filter out the placeholder comment on the text box 'New Comment'

**Comment by Ortiz-Sergio** [comment]
I decided to only filter out unwanted comments, without giving the user a chance to enter their comment again since the website reloads after entering a comment anyways, so users can try again. Tell me if anything about this approach isn't valid, or if there is a better way to do it.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 2 above)
Comment (Ortiz-Sergio) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:77]: Makes sense!

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -52,4 +75,20 @@ private String convertToJsonUsingGson(List<String> comments) {
...
+      if (comment.isEmpty()) {
+          return false;
+      }
+      if (comment.isBlank()) {
+          return false;
+      }
+      if (comment.equalsIgnoreCase("New Comment")) {
+          return false;
+      }
+      return true;
```
Comment (abhijitnukalapati) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: minor - you could also just do something succinct:
`return !comment.isEmpty() && !comment.isBlank() && !comment.equalsIgnoreCase("New Comment")`

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 4]
```diff
@@ -28,21 +31,41 @@
...
   }
 
-  private void initializeComments() {
-    commentList.add("Awesome Website!");
-    commentList.add("Great job! Keep it up!");
-    commentList.add("Very interesting!!");
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+      String comment = getParameter(request, "comment-input", "");
+      if(isValid(comment)) {
```
Comment (abhijitnukalapati) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:63]: come to think of it, can we do some input validation in the UI? That way we have a more responsive web app

File: portfolio/src/main/webapp/index.html [hunk 5]
```diff
@@ -195,9 +195,20 @@ <h2 style="color:FloralWhite; background-color:green;">
...
+      <p>Enter your comment: </p>
+      <textarea name="comment-input">New Comment</textarea>
+      <br/><br/>  
+      <input type="submit" />
+      <br/><br/>
+      </form>
+
       <div style="border: 1px solid">
-      <button onclick="getServerComments()">Comments</button>
+        <button onclick="getServerComments()">Comments</button>
```
Comment (abhijitnukalapati) [inline portfolio/src/main/webapp/index.html]: _nitpick_ - remove indentation (this is admittedly minor, but as these keep adding up, the code can start to look messy)

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 3 above)
Comment (Ortiz-Sergio) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: True!

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 4 above)
Comment (Ortiz-Sergio) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:63]: Sounds good! I'll display a little message prompting the user to try again

File: portfolio/src/main/webapp/index.html (hunk 5 above)
Comment (Ortiz-Sergio) [reply portfolio/src/main/webapp/index.html]: Yes! I also dislike these but I am having trouble spotting them on the editor for some reason

**Comment by Ortiz-Sergio** [comment]
I added some input validation where a little text box pops up if the user enters an invalid comment. I added it on the html though so I am not sure if it is overkill to do validation both on the html and on the java servlet. If so, I could remove the one on teh java servlet since by the time a comment gets in it will already be a valid one.

**Comment by abhijitnukalapati** [comment]
> I added some input validation where a little text box pops up if the user enters an invalid comment. I added it on the html though so I am not sure if it is overkill to do validation both on the html and on the java servlet. If so, I could remove the one on teh java servlet since by the time a comment gets in it will already be a valid one.


That's true. Good point. We can remove the validation on the servlet

File: portfolio/src/main/webapp/index.html [hunk 6]
```diff
@@ -195,7 +195,29 @@ <h2 style="color:FloralWhite; background-color:green;">
...
+          button at the bottom of the page to see some other comments!
+      </p>
+
+      <script>
+      function validateForm() {
+          var comment = document.forms["commentSection"]["comment-input"].value;
+          var isBlank = comment.toUpperCase() === "NEW COMMENT";
+          if (isBlank || !comment.replace(/\s/g, '').length) {
+              alert("Not a valid comment, try again");
+              return false;
```
Comment (abhijitnukalapati) [inline portfolio/src/main/webapp/index.html:208]: looks good!

**Pull request merged**